pkg/gke/services/gke.go

```diff
@@ -9,110 +9,98 @@ import (
 )
 
 type GKEClusterService interface {
-	Create(parent string, cluster *gkeapi.CreateClusterRequest) (*gkeapi.ProjectsLocationsClustersCreateCall, error)
-	List(parent string) (*gkeapi.ProjectsLocationsClustersListCall, error)
-	Get(name string) (*gkeapi.ProjectsLocationsClustersGetCall, error)
-	Delete(name string) (*gkeapi.ProjectsLocationsClustersDeleteCall, error)
-	SetNetworkPolicy(name string, networkPolicy *gkeapi.SetNetworkPolicyRequest) (*gkeapi.ProjectsLocationsClustersSetNetworkPolicyCall, error)
-	SetMaintenancePolicy(name string, maintenancePolicy *gkeapi.SetMaintenancePolicyRequest) (*gkeapi.ProjectsLocationsClustersSetMaintenancePolicyCall, error)
-	SetResourceLabels(name string, resourceLabels *gkeapi.SetLabelsRequest) (*gkeapi.ProjectsLocationsClustersSetResourceLabelsCall, error)
+	ClusterCreate(ctx context.Context, parent string, createclusterrequest *gkeapi.CreateClusterRequest) (*gkeapi.Operation, error)
+	ClusterList(ctx context.Context, parent string) (*gkeapi.ListClustersResponse, error)
+	ClusterGet(ctx context.Context, name string) (*gkeapi.Cluster, error)
+	ClusterUpdate(ctx context.Context, name string, updateclusterrequest *gkeapi.UpdateClusterRequest) (*gkeapi.Operation, error)
+	ClusterDelete(ctx context.Context, name string) (*gkeapi.Operation, error)
+	SetNetworkPolicy(ctx context.Context, name string, networkpolicyrequest *gkeapi.SetNetworkPolicyRequest) (*gkeapi.Operation, error)
+	SetMaintenancePolicy(ctx context.Context, name string, maintenancepolicyrequest *gkeapi.SetMaintenancePolicyRequest) (*gkeapi.Operation, error)
+	SetResourceLabels(ctx context.Context, name string, resourcelabelsrequest *gkeapi.SetLabelsRequest) (*gkeapi.Operation, error)
+	NodePoolCreate(ctx context.Context, parent string, createnodepoolrequest *gkeapi.CreateNodePoolRequest) (*gkeapi.Operation, error)
+	NodePoolList(ctx context.Context, parent string) (*gkeapi.ListNodePoolsResponse, error)
+	NodePoolGet(ctx context.Context, name string) (*gkeapi.NodePool, error)
+	NodePoolUpdate(ctx context.Context, name string, updatenodepoolrequest *gkeapi.UpdateNodePoolRequest) (*gkeapi.Operation, error)
+	NodePoolDelete(ctx context.Context, name string) (*gkeapi.Operation, error)
+	SetSize(ctx context.Context, name string, setnodepoolsizerequest *gkeapi.SetNodePoolSizeRequest) (*gkeapi.Operation, error)
+	SetAutoscaling(ctx context.Context, name string, setnodepoolautoscalingrequest *gkeapi.SetNodePoolAutoscalingRequest) (*gkeapi.Operation, error)
+	SetManagement(ctx context.Context, name string, setnodepoolmanagementrequest *gkeapi.SetNodePoolManagementRequest) (*gkeapi.Operation, error)
 }
 
 type gkeClusterService struct {
-	svc *gkeapi.ProjectsLocationsClustersService
+	svc gkeapi.Service
 }
 
-func NewGKEClusterService(ctx context.Context, ts oauth2.TokenSource) (GKEClusterService, error) {
+func NewGKEClusterService(ctx context.Context, ts oauth2.TokenSource) (*gkeClusterService, error) {
```

Comment:
Shouldn’t this return interface as the structure implements it?

Reply:
It is good question, but aks is also returning this
https://github.com/rancher/aks-operator/blob/master/pkg/aks/services/workplaces.go#L20